testing/sdk_tasks.py

```diff
@@ -57,16 +57,21 @@ def fn():
         prefix_clause = ''
         if prefix:
             prefix_clause = ' starting with "{}"'.format(prefix)
-        log.info('Waiting for {} tasks{} to have updated ids:\n- Old tasks: {}\n- Current tasks: {}'.format(
-            len(old_task_ids), prefix_clause, sorted(old_task_ids), sorted(task_ids)))
-
-        all_updated = True
-        for id in task_ids:
-            if id in old_task_ids:
-                all_updated = False
-        if len(task_ids) < len(old_task_ids):
-            all_updated = False
-        return all_updated
+
+        old_set = set(old_task_ids)
+        new_set = set(task_ids)
+        newly_launched_set = (new_set - old_set)
+        all_updated = len(newly_launched_set) == len(new_set)
+        if all_updated:
+            return all_updated
+
+        # forgive the language a bit, but len('remained') == len('launched'),
+        # and similar for the rest of the label for task ids in the log line,
+        # so makes for easier reading
+        log.info('Waiting for tasks{} to have updated ids:\n- Old tasks (remained): {}\n- New tasks (launched): {}'.format(
+            prefix_clause,
+            old_set & new_set,
```

Comment:
I find old_set.intersection(new_set) much clearer, but ignore if you don't.

Reply:
See above, set math should be known. To further argue the point, Python didn't provide `&&` for something so commonly used `if A && B:`, but did provide these set symbols.

Reply:
I do not know what `old_set & new_set` means.  I had to look it up three times while reading this code, because I repeatedly forgot.   If you feel my knowledge is insufficient, then that's OK, but I didn't provide this feedback just to pick nits.

Reply:
Understood and will change. I don't set the bar, and your feedback is greatly appreciated. If set math isn't clear, set methods will be used.